### ISSUE TITLE
Error: Problem with `filter()` input `..1`

### ISSUE THREAD

**Issue by Svenja-vS** [chrisschuerz/SWATplusR#23]
Hello everyone,

I am using SWATplusR for some month now and was able to calibrate a model-project successfully. Recently I included some management operations in a model, which was calibrated and working in SWATplusR before and now I got the following error message:

`Error: Problem with `filter()` input `..1`.
x object 'run_num' not found
i Input `..1` is `!is.na(run_num)`.`

I am now not even able to run the previous model that was working. I tried also to run the model on another computer with SWATplusR, but I got the same error message.

What I also tried, was to see if the demo version is still working and it is!
Hope some can help me here and if you need more information please let me know.

Thanks in advance 
Svenja

**Comment by Svenja-vS** [comment]
Hello again,
More specific the >safe file< comment in the code is causing the error. Below I included the basic run_swat2012 function I am using and I am not able to save my simulation, as I was before in the first version of my model.
`q_sim_test <- run_swat2012(project_path = project_sdc2,
                           output = define_output(file = "rch",
                                                  variable = "FLOW_OUT",
                                                  unit = 3),
                           parameter = par_mod,
                           start_date = '2009-01-01',
                           end_date = '2015-12-31',
                           years_skip = 3,
                           save_file = "q_sim_sdc2_20200831_0812")`

Maybe this is helping - I still can not figure out what exactly is causing the error or how to solve the problem.

Thanks, Svenja

**Comment by WillemVervoort** [comment]
@Svenja-vS Can you share par_mod?

**Comment by chrisschuerz** [comment]
Hi Svenja,

I tried to reporduce your error with the demo SWAT2012 rev622 model. I tested the following piece of code:

```r
library(SWATplusR)
library(tibble)

pth <- "C:/swat2012_rev622_demo"


par_mod <- tibble("CN2.mgt|change = abschg" = seq(0, 7.5, 2.5))

q_sim_test <- run_swat2012(project_path = "Path:/to_my_SWAT2012_demo", 
                           output = define_output(file = "rch", variable = "FLOW_OUT", unit = 3), 
                           parameter = par_mod, 
                           start_date = '2009-01-01', 
                           end_date = '2012-12-31', 
                           years_skip = 1,
                           save_file = "test")
```
This example is quite comparable to your code. It worked on my machine with `SWATplusR` version 0.2.8.9010 (I think it should also work with older versions, as there were nor changes in the save routine). 

Could you also maybe provide the following info:
- The version number of `SWATplusR` on your machine 
- The SWAT2012 revision number (maybe you use a version that has a different output file format, that couses now troubles when reading).
- Are you writing a new `save_file` or dou you want to write into an existing save folder?

Cheers 
Christoph

**Comment by Svenja-vS** [comment]
Hello again,

Thanks for the quick reply.
Yes @WillemVervoort.  I am using as input for par_mod the following ...

`par_mod <- tibble ("cn2_pinh::CN2.mgt|change = absval | luse %in% 'PINH'" = 35,
                   "cn2_euco::CN2.mgt|change = absval | luse %in% 'EUCO'" = 35,
                   "cn2_eucx::CN2.mgt|change = absval | luse %in% 'EUCX'" = 35,
                   "cn2_eucb::CN2.mgt|change = absval | luse %in% 'EUCB'" = 35,
                   "cn2_eucc::CN2.mgt|change = absval | luse %in% 'EUCC'" = 35,
                   "SOL_AWC.sol|change = absval" = 0.45,
                   "GW_DELAY.gw|change = absval" = 2,
                   "GWQMN.gw|change = absval" = 6000,
                   "ESCO.hru|change = absval " = 0.025,
                   "RCHRG_DP.gw|change = absval" = 0.15,
                   "REVAPMN.gw|change = absval" = 750,
                   "GW_REVAP.gw|change = absval" = 0.02,
                   "SURLAG.bsn|change = absval" = 4,
                   "ALPHA_BF.gw|change = absval"= 0.048)`

hope it might helps.

Also @chrisschuerz I am using the SWATplusR version 0.2.7
and SWAT2012 revision number 2012.10_5.21.
I am saving the output in a existing folder. 
I still not sure why I have troubles by saving the file, since it was working before but but well.
I could also try to save the demo version, which I did not try yet. Let you know.

Svenja

**Comment by Svenja-vS** [comment]
In addition, when I try to save the demo SWAT2012 rev622 model this is not working as well.
Svenja

**Comment by chrisschuerz** [comment]
Ah interesting. It worked on my computer but with `SWATplusR` version 0.2.8.9010. Could you update the R package and retry it?

just remove `SWATplusR` from your packages and reinstall from GitHub. This should do the trick:

```r
remove.packages('SWATplusR')
devtools::install_github('chrisschuerz/SWATplusR')
```

The most recent version should be 0.2.8.2011 then.

**Comment by NataljaC** [comment]
Hello,
I had encountered the same issue with run_swatplus(). After painstaking hours of ... yoga and meditation.. I found a workaround, Just add:
 `run_num <- 1 ` 
before the run_swat_X()
It is a temporary fix, and I hope that the bug will surface soon.
Best,
Natalja

**Comment by Svenja-vS** [comment]
Hello,

Good news it is working again. I did the update and with the recent version it is working but I had to change my RStudio version from 4.0.2 to 3.6.0, because SWATplusR is not compatible with the 4.0.2 version of R. Before I had no troubles with that, but okay it is working so :) Thanks.
@NataljaC thanks for the help as well, in case I will have some similar issues again I will try ;)

By Svenja

**Comment by chrisschuerz** [comment]
@Svenja-vS thanks for the update. This would have been my next question that I forgot to ask in my previous posts, which R version you are using. I did not find time yet, but the update of `SWATplusR` to R 4.x is planned in the next 1-2 weeks. 
I hope it is OK for your work to stay with R 3.6.x at the moment.

@NataljaC, thank you for posting the work around here. I will in any case look into the issue. Which R version are you using?

Cheers
Christoph

**Comment by NataljaC** [comment]
Indeed, I was using the newest version of R. I did the installations from
scratch just two-three weeks ago.
I will now revert to the old versions and try out things.
Cheers,
Natalja

**Comment by Svenja-vS** [comment]
Yes no worries, for me it's fine to switch the version in R for now.
Thanks for help 
Greetings,
Svenja

**Comment by chrisschuerz** [comment]
Hi guys,

it took my way too long to look into this issue. Well but what else would I do on a Saturday evening these days... 🙈 
The issue should be fixed with the most recent version 0.3.3.9003. If you are still interested in using the saving to sqlite option with SWATplusR please update the R package and try it.

Cheers 
Christoph
